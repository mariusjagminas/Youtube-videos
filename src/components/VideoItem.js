import React from "react";
import { withStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import Divider from "@material-ui/core/Divider";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import PropTypes from "prop-types";

const styles = theme => ({
  gridContainer: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },

  imgContainer: {
    width: "100%",
    paddingBottom: "56.25%",
    height: 0,
    overflow: "hidden"
  },
  img: {
    width: "100%"
  }
});

const VideoItem = ({
  imgUrl,
  title,
  channelTitle,
  videoId,
  index,
  classes,
  changeMainVideo
}) => {
  return (
    <>
      <ListItem button onClick={changeMainVideo.bind(null, index)}>
        <Grid container spacing={2} className={classes.gridContainer}>
          <Grid item xs={8}>
            <div className={classes.imgContainer}>
              <img className={classes.img} alt={title} src={imgUrl} />
            </div>
          </Grid>
          <Grid item xs={4}>
            <Typography component="h2" variant="body1" color="textPrimary">
              {title}
            </Typography>
            <Typography component="div" variant="body2" color="textSecondary">
              {channelTitle}
            </Typography>
          </Grid>
        </Grid>
      </ListItem>
      <Divider />
    </>
  );
};

export default withStyles(styles)(VideoItem);

VideoItem.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  channelTitle: PropTypes.string.isRequired,
  videoId: PropTypes.string.isRequired
};
