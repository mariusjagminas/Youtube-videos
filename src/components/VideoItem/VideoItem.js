import React from "react";
import { withStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import Divider from "@material-ui/core/Divider";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import PropTypes from "prop-types";

const styles = theme => ({
  gridContainer: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },

  imgContainer: {
    width: "100%",
    paddingBottom: "56.25%",
    height: 0,
    overflow: "hidden"
  },
  img: {
    width: "100%"
  }
});

const VideoItem = ({
  imgUrl,
  title,
  channelTitle,
  index,
  classes,
  changeMainVideo
}) => {
  return (
    <>
      <ListItem
        button
        onClick={changeMainVideo.bind(null, index)}
        data-test="button"
      >
        <Grid container spacing={2} className={classes.gridContainer}>
          <Grid item xs={8}>
            <div className={classes.imgContainer}>
              <img className={classes.img} alt={title} src={imgUrl} />
            </div>
          </Grid>
          <Grid item xs={4}>
            <Typography
              noWrap={true}
              component="h2"
              variant="body1"
              color="textPrimary"
            >
              {title}
            </Typography>
            <Typography
              noWrap={true}
              component="div"
              variant="body2"
              color="textSecondary"
            >
              {channelTitle}
            </Typography>
          </Grid>
        </Grid>
      </ListItem>
      <Divider />
    </>
  );
};

export default withStyles(styles)(VideoItem);

VideoItem.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  channelTitle: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  changeMainVideo: PropTypes.func.isRequired
};
