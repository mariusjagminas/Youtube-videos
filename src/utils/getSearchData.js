import axios from "axios";
import { config } from "../config";

const url = "https://www.googleapis.com/youtube/v3/search";

export const getSearchData = async term => {
  try {
    const response = await axios.get(url, {
      params: {
        part: "snippet",
        maxResults: 5,
        key: config.API_KEY,
        q: term
      }
    });
    return response.data.items;
  } catch {}
};

// https://www.youtube.com/embed/ video id
