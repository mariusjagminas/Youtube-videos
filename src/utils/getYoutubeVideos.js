import axios from "axios";

const url = "https://www.googleapis.com/youtube/v3/search";

export const getYoutubeVideos = async term => {
  try {
    const response = await axios.get(url, {
      params: {
        part: "snippet",
        maxResults: 5,
        key: process.env.REACT_APP_YOUTUBE_KEY,
        q: term
      }
    });
    // excluding playlists and channels items from returned list
    const videos = await response.data.items.filter(vid => vid.id.videoId);

    return { videos: videos, error: null };
  } catch (err) {
    return { videos: [], error: err.message };
  }
};
