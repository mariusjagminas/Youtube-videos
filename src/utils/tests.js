import renderer from "react-test-renderer";
import propTypesFunc from "check-prop-types";

export const checkIfMatchSnapshot = component => {
  return it("should match a snapshot", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const checkPropTypes = (component, props) => {
  return it("should have correct propTypes", () => {
    const result = propTypesFunc(
      component.propTypes,
      props,
      "props",
      component.name
    );
    expect(result).toBe(undefined);
  });
};
