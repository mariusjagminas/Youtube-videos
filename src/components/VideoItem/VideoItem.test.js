import React from "react";
import { checkIfMatchSnapshot } from "../../utils/tests";
import { shallow } from "enzyme";
import VideoItem from "./VideoItem";

describe("<VideoItem/>", () => {
  const props = {
    imgUrl: "test-g6373.img",
    title: "test-title-1245766",
    channelTitle: "test-channel-76478347834",
    index: 1,
    changeMainVideo: jest.fn()
  };

  const wrapper = shallow(<VideoItem {...props} />).dive();

  it("should render a video thumbnail", () => {
    expect(
      wrapper.containsMatchingElement(<img src="test-g6373.img" />)
    ).toBeTruthy();
  });

  it("should render the video title", () => {
    expect(wrapper.contains("test-title-1245766")).toBeTruthy();
  });

  it("should render the channel title", () => {
    expect(wrapper.contains("test-channel-76478347834")).toBeTruthy();
  });

  it("should call changeMainVideo function, and pass an index number as the arg in it when clicked", () => {
    const shallowWrapper = wrapper.find('[data-test="button"]');
    shallowWrapper.simulate("click");

    expect(props.changeMainVideo.mock.calls[0][0]).toBe(1);
  });

  checkIfMatchSnapshot(<VideoItem {...props} />);
});
