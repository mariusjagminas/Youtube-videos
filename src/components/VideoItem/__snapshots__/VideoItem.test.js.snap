// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem component should match a snapshot 1`] = `
Array [
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    data-test="button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiGrid-root VideoItem-gridContainer-1 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          className="VideoItem-imgContainer-2"
        >
          <img
            alt="test-title-1245766"
            className="VideoItem-img-3"
            src="test-g6373.img"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <h2
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-noWrap"
        >
          test-title-1245766
        </h2>
        <div
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap"
        >
          test-channel-76478347834
        </div>
      </div>
    </div>
  </div>,
  <hr
    className="MuiDivider-root"
  />,
]
`;
